### PR TITLE
Add PBS emulation support

### PR DESCRIPTION
**Pull request by milljm**
I have the SJS module available in what will be our latest moose-environment package. But this must be included into MOOSE before we release the actual emulator.

Our goal was to make the TestHarness unaware of any emulation present, so very little changes were actually needed in our TestHarness.

Refs #5746

### CODE REVIEW

File: python/TestHarness/TestHarness.py [hunk 1]
```diff
@@ -468,7 +468,7 @@ def buildPBSBatch(self, output, tester):
       return ('QSUB NOT FOUND', '')
     else:
       # Get the PBS Job ID using qstat
-      results = re.findall(r'JOB_NAME: (\w+\d+) JOB_ID: (\d+) TEST_NAME: (\S+)', output, re.DOTALL)
+      results = re.findall(r'JOB_NAME: (\w+) JOB_ID:.* (\d+).*TEST_NAME: (\S+)', output, re.M)
```
Comment (permcody) [inline python/TestHarness/TestHarness.py]: This change is slightly suspect but so was the original line upon closer inspection. I really don't think you need the "M" option at all since there are no anchors (^,$) in this expression. However, losing the "DOTALL" may or may not be the right thing here. With it if the Pattern repeats, it's possible to have the "dot star" operator match part of an earlier pattern with the ending of a latter pattern. That operator is greedy so that is what normally happens. With it turned off though, your "dot star" operators can never span more than a single line because dot doesn't match newlines by default. 

TL;DR
I suspect the right answer is losing both "M" and "DOTALL".

**Comment by milljm** [comment]
Do not merge this... I have to clean it up a bit.

File: python/TestHarness/TestHarness.py (hunk 1 above)
Comment (milljm) [inline python/TestHarness/TestHarness.py]: I think... I was _thinking_ I was being smart... I know I need the regex changes, but not the global regex modifier. I probably thought DOTALL did -not- include multilines, when the cluster_launcher is returning them:
```
cluster_launcher.py pbs_001.cluster
gold
JOB_NAME: simple_00_004 JOB_ID: 244367 TEST_NAME: kernels/simple_diffusion.test
```

In any case, I am cleaning up this junk and will have a stronger pull request shortly!

**Comment by permcody** [comment]
Closed in favor of #6110
